src/main/kotlin/interpreter/pyobject/PyDict.kt

```diff
@@ -31,27 +31,20 @@ class PyDict(val items: LinkedHashMap<out PyObject, out PyObject>) : PyObject(Py
          * Creates a new PyDict from any map, wrapping primitive types.
          */
         fun fromAnyMap(map: Map<*, *>): PyDict {
-            val newMap = map.map {
-                val key = if (it.key !is PyObject) {
-                    wrapPrimitive(it.key)
-                } else {
-                    it.key as PyObject
-                }
-
-                val value = if (it.value !is PyObject) {
-                    wrapPrimitive(it.value)
-                } else {
-                    (it.value as PyObject)
-                }
-
-                Pair(key, value)
-            }.toMap().toMutableMap()
-            return PyDict(newMap as LinkedHashMap<out PyObject, out PyObject>)
+            val newMap = map.entries.associateByTo(
+                linkedMapOf<PyObject, PyObject>(),
+                {
+                    it.key as? PyObject ?: wrapPrimitive(it.key)
+                }, {
+                    it.value as? PyObject ?: wrapPrimitive(it.value)
+                })
```

Comment:
I prefer my solution, this is just kinda ugly to read - plus I think the perf impact of creating Pair() will be mitigated be it by JIT magic or Valhalla in the future.